fix positioning of nested tspans

HackbrettXXX opened the pull request:
This PR addresses #1548. The fix seems to be to just remove the lines where the `x/y/dx/dy` attributes are copied from the parent text element to the first child. If I'm correct, the code is not required, because the current text position for the first child is already applied in `initializeCoordinates`. I've checked with the other test cases and they seem to still work. @gabelerner do you happen to remember why the code is there? Am I missing something?

The new test case will probably fail on CI, because the snapshot was created on a Windows machine. The tests didn't pass on my Ubuntu VM, either, so I couldn't use it to create the snapshots. Could you maybe re-create the snapshots?

dangreen commented:
@HackbrettXXX Hi. Tests are failed. Your changes shouldn't affect other test cases.

HackbrettXXX commented:
Yes, I'm going to look into that.

dangreen commented:
@HackbrettXXX [Here](https://github.com/canvg/canvg/actions/runs/1553516366) you can download artifacts with diff snapshots.

HackbrettXXX commented:
Ah, thanks. Seems like I've gotten a step closer. Only the text-anchor seems to be broken now. I'll have a look at it.

HackbrettXXX commented:
Alright, I've fixed the issues (except for the one test case in node, see below). I also took the chance to improve output for RTL text a bit. It wasn't working before as soon as there were multiple tspans. I also fixed the detection of Arabic terminal and medial glyphs.

The failing test in node is a bit weird, because it works in the browser. It seems like the `ctx.save/restore` calls are not balanced or don't work at all. The original snapshot was also broken:

Node:
![node-issue38 svg-snap](https://user-images.githubusercontent.com/16103654/145432145-f27250d9-47f5-4a0c-9a53-d6e7a7309660.png)

Browser:
![offscreen-browser-issue38 svg-snap](https://user-images.githubusercontent.com/16103654/145432325-b7181ea8-030b-4b1e-8770-9a09fcecb00c.png)

Current diff in Node:
![node-issue38 svg-diff](https://user-images.githubusercontent.com/16103654/145432740-04e54660-87e7-4de8-9357-79038618e545.png)

The font-weight isn't restored to the original value after the tspan. Could this be due to a broken canvas polyfill in node?

dangreen commented:
@HackbrettXXX Ok, if original node snapshot is also broken, then let's just update this snapshot to pass tests.

gabelerner commented:
@HackbrettXXX nice job getting it all to pass and validating the assumptions
@dangreen ping on reviewing this ... i'll take a look in more detail sometime this upcoming week.

dangreen commented on src/Document/TextElement.ts:
```diff
@@ -9,16 +9,39 @@ import {
 
 export default class TextElement extends RenderedElement {
+	private static inheritAttribute(child: TextElement, name: string): string | null {
+		function isFirstChild(child: TextElement): boolean {
```
Let's move it to [Element](https://github.com/canvg/canvg/blob/master/src/Document/Element.ts)

```js
child.isFirstChild()
```

dangreen commented on src/Document/TextElement.ts:
```diff
@@ -9,16 +9,39 @@ import {
 import RenderedElement from './RenderedElement';
 
 export default class TextElement extends RenderedElement {
+	private static inheritAttribute(child: TextElement, name: string): string | null {
```
Let's rename it to `getInheritAttribute`

dangreen commented on src/Document/TextElement.ts:
```diff
@@ -337,53 +382,57 @@ export default class TextElement extends RenderedElement {
+			// are only inherited from a parent to it's first child.
 			if (!xAttr.hasValue()) {
-				xAttr.setValue(textParent.getAttribute('x').getValue('0'));
+				xAttr.setValue(TextElement.inheritAttribute(child, 'x'));
```
Is child can be not `TextElement`? Is `getInheritAttrubute` must be a static method?

HackbrettXXX replied:
I think at this place `child` is always a `TextElement` because of the
```
if (typeof child.measureText !== 'function') {
  return child;
}
```
check before. I've converted it to an instance method now.

dangreen commented:
@HackbrettXXX Thank you for contribution!

dangreen commented:
@HackbrettXXX released in https://www.npmjs.com/package/canvg/v/3.0.10
